lib/api/apiUtils/authorization/validation.js

```diff
@@ -0,0 +1,217 @@
+            return cb(err);
+        }
+        if (bucketShield(bucket, requestType)) {
+            log.debug('bucket is shielded from request', {
+                requestType,
+                method: 'metadataGetBucketAndObject',
+            });
+            // still return bucket for cors headers
+            return cb(errors.NoSuchBucket, bucket);
+        }
+        log.trace('found bucket in metadata');
+        return cb(null, bucket);
+    });
+}
+
+/** metadataValidateBucket - retrieve bucket from metadata and check if user
+ * is authorized to access it
+ * @param {object} params - function parameters
+ * @param {AuthInfo} params.authInfo - AuthInfo class instance, requester's info
+ * @param {string} params.bucketName - name of bucket
+ * @param {string} params.requestType - type of request
+ * @param {RequestLogger} log - request logger
+ * @param {function} callback - callback
+ * @return {undefined} - and call callback with params err, bucket md
+ */
+export function metadataValidateBucket(params, log, callback) {
+    const { authInfo, bucketName, requestType } = params;
+    const canonicalID = authInfo.getCanonicalID();
+    return metadataGetBucket(requestType, bucketName, log, (err, bucket) => {
+        if (err) {
+            // still return bucket for cors headers
+            return callback(err, bucket);
+        }
+        if (!isBucketAuthorized(bucket, requestType, canonicalID)) {
+            log.debug('access denied for user on bucket', { requestType });
+            return callback(errors.AccessDenied, bucket);
+        }
+        return callback(null, bucket);
+    });
+}
+
+/** getNullVersion - return metadata of null version if it exists
+ * @param {object} objMD - metadata of master version
+ * @param {string} bucketName - name of bucket
+ * @param {string} objectKey - name of object key
+ * @param {RequestLogger} log - request logger
+ * @param {function} cb - callback
+ * @return {undefined} - and call callback with params err, objMD of null ver
+ */
+export function getNullVersion(objMD, bucketName, objectKey, log, cb) {
+    const options = {};
+    if (objMD.isNull || !objMD.versionId) {
+        // null version is current version
+        log.debug('found null version'); // don't want to log internal id?
+        return cb(null, objMD);
+    }
+    if (objMD.nullVersionId) {
+        // the latest version is not the null version, but null version exists
+        log.debug('null version exists, get the null version');
+        options.versionId = objMD.nullVersionId;
+        return metadata.getObjectMD(bucketName, objectKey, options, log, cb);
+    }
+    log.debug('could not find a null version');
+    return cb();
+}
+
+/** metadataGetBucketAndObject - retrieves bucket and specified version
+ * NOTE: This function only returns master version if 'null' version is
+ * of object is specified. Getting the actual null version must be a manual
```

Comment:
if trying to get the null version instead of any other version using this function, specified in the param `versionId`. 'Actual null version' --> instead of the master version objMD.
I'll rewrite the note to be more clear